packages/dev/core/src/Inputs/scene.inputManager.ts

```diff
@@ -392,18 +406,24 @@ export class InputManager {
             }
         }
 
-        if (pickResult) {
-            const type = PointerEventTypes.POINTERDOWN;
+        let pointerInfo: PointerInfo;
+        const type = PointerEventTypes.POINTERDOWN;
 
-            if (scene.onPointerDown) {
-                scene.onPointerDown(evt, pickResult, type);
-            }
+        if (scene.onPointerDown) {
+            pickResult = pickResult ? pickResult : this._pick(evt.pointerId, type);
+            // We know that _pick will return a PickingInfo because _internalPick always returns a PickingInfo
+            scene.onPointerDown(evt, pickResult, type);
+        }
 
-            if (scene.onPointerObservable.hasObservers()) {
-                const pi = new PointerInfo(type, evt, pickResult);
-                this._setRayOnPointerInfo(pickResult, evt);
-                scene.onPointerObservable.notifyObservers(pi, type);
-            }
+        if (pickResult) {
+            pointerInfo = new PointerInfo(type, evt, pickResult);
+            this._setRayOnPointerInfo(pickResult, evt);
+        } else {
+            pointerInfo = new PointerInfo(type, evt, null, this);
+        }
+
+        if (scene.onPointerObservable.hasObservers()) {

```

Comment:
`pointerInfo` is now created regardless of this if statement. Is that correct?

Reply:
I'm just noting that it's different than [before](https://github.com/BabylonJS/Babylon.js/pull/13145/files/06d5df92468766b1681046560637aa7b3e4f90fa#diff-745b65ff5eafe824c3c6d1595955b9eb97bcfa6cb29a90b084f1d7425336e4c8L402-L406).

Reply:
Technically, in the older version of the code, if there was an attached pointer input, we'd always create a PointerInfo anyway so the only difference would be the if check for a pickResult to determine if we're providing an inputManager ref.  Given how often down events occur, any perf hit would be negligible.

Reply:
Yes, but now it will create a PointerInfo even if there are no observers. But I agree there isn't a big issue.